nixos/modules/services/networking/vpp.nix

```
@@ -0,0 +1,543 @@
+                    default = [ ];
+                    description = "Device types to blacklist, using PCI vendor:device syntax.";
+                  };
+                };
+
+                cpu = {
+                  main-core = mkOption {
+                    type = types.nullOr types.ints.unsigned;
+                    example = 0;
+                    default = null;
+                    description = ''
+                      Logical CPU core where the main thread runs, if unset VPP will use core 1 if available.
+                    '';
+                  };
+
+                  corelist-workers = mkOption {
+                    type = types.nullOr types.str;
+                    example = "2-3,18-19";
+                    default = null;
+                    description = ''
+                      Explicitly set logical CPUs on which VPP worker threads will run.
+
+                      {option}`skip-cores` and {option}`workers` are incompatible with {option}`corelist-workers`.
+                    '';
+                  };
+
+                  skip-cores = mkOption {
+                    type = types.nullOr types.ints.positive;
+                    example = 8;
+                    default = null;
+                    description = ''
+                      Set number of logical CPU cores to skip when using the {option}`workers` option.
+
+                      {option}`skip-cores` and {option}`workers` are incompatible with {option}`corelist-workers`.
+                    '';
+                  };
+                  workers = mkOption {
+                    type = types.nullOr types.ints.positive;
+                    example = 4;
+                    default = null;
+                    description = ''
+                      Set number of workers to be created and automatically assigned. Workers will be pinned to
+                      N consecutive CPU cores while skipping {option}`skip-cores` CPU core(s) and main thread's core.
+
+                      {option}`skip-cores` and {option}`workers` are incompatible with {option}`corelist-workers`.
+                    '';
+                  };
+                };
+              };
+          };
+        };
+
+        settingsFile = mkOption {
+          type = types.path;
+          example = "/etc/vpp/startup.conf";
+          default = pkgs.writeText "vpp-${name}.conf" (parseConfig cfg.instances.${name}.settings);
+          defaultText = ''
+            pkgs.writeText "vpp-\${name}.conf" (parseConfig config.services.vpp.instances.\${name}.settings);
+          '';
+          description = ''
+            Configuration file passed to {command}`vpp -c`.
+            It is recommended to use the {option}`settings` option instead.
+
+            Setting this option will override the config file
+            auto-generated from the {option}`settings` option.
+          '';
+        };
+
+        startupConfig = mkOption {
+          type = types.str;
```

```suggestion
          type = types.lines;
```
otherwise the module system cannot merge the option